src/Psalm/Internal/Analyzer/Statements/Block/LoopAnalyzer.php

```diff
@@ -554,10 +698,7 @@ private static function updateLoopScopeContexts(
         }
 
         // merge vars possibly in scope at the end of each loop
-        $loop_context->vars_possibly_in_scope = array_merge(
-            $loop_context->vars_possibly_in_scope,
-            $loop_scope->vars_possibly_in_scope,
-        );
+        $loop_context->vars_possibly_in_scope += $loop_scope->vars_possibly_in_scope;
```

Comment:
Not sure about this one. `array_merge()` would overwrite `$loop_context->vars_possibly_in_scope['a']` with `$loop_scope->vars_possibly_in_scope['a']`, but `+=` would keep the value of `$loop_context->vars_possibly_in_scope['a']`